examples/stress_tests/bevymark.rs

```diff
@@ -133,115 +238,247 @@ fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
-            .nth(1)
-            .and_then(|arg| arg.parse::<usize>().ok())
-            .unwrap_or_default(),
-        wave: std::env::args()
-            .nth(2)
-            .and_then(|arg| arg.parse::<usize>().ok())
-            .unwrap_or(1),
-    });
+    let mut scheduled = BirdScheduled {
+        per_wave: args.per_wave,
+        waves: args.waves,
+    };
+
+    if args.benchmark {
+        let counter = counter.into_inner();
+        for wave in (0..scheduled.waves).rev() {
+            spawn_birds(
+                &mut commands,
+                args,
+                &windows.single().resolution,
+                counter,
+                scheduled.per_wave,
+                &mut bird_resources,
+                Some(wave),
+                wave,
+            );
+        }
+        scheduled.waves = 0;
+    }
+    commands.insert_resource(bird_resources);
+    commands.insert_resource(scheduled);
 }
 
+#[allow(clippy::too_many_arguments)]
 fn mouse_handler(
     mut commands: Commands,
+    args: Res<Args>,
     time: Res<Time>,
     mouse_button_input: Res<Input<MouseButton>>,
     windows: Query<&Window>,
-    bird_texture: Res<BirdTexture>,
+    bird_resources: ResMut<BirdResources>,
     mut counter: ResMut<BevyCounter>,
+    mut rng: Local<Option<StdRng>>,
+    mut wave: Local<usize>,
 ) {
+    if rng.is_none() {
+        *rng = Some(StdRng::seed_from_u64(42));
+    }
+    let rng = rng.as_mut().unwrap();
     let window = windows.single();
 
     if mouse_button_input.just_released(MouseButton::Left) {
-        let mut rng = thread_rng();
         counter.color = Color::rgb_linear(rng.gen(), rng.gen(), rng.gen());
     }
 
     if mouse_button_input.pressed(MouseButton::Left) {
         let spawn_count = (BIRDS_PER_SECOND as f64 * time.delta_seconds_f64()) as usize;
         spawn_birds(
             &mut commands,
+            args.into_inner(),
             &window.resolution,
             &mut counter,
             spawn_count,
-            bird_texture.clone_weak(),
+            bird_resources.into_inner(),
+            None,
+            *wave,
         );
+        *wave += 1;
     }
 }
 
+fn bird_velocity_transform(
+    half_extents: Vec2,
+    mut translation: Vec3,
+    velocity_rng: &mut StdRng,
+    waves: Option<usize>,
+    dt: f32,
+) -> (Transform, Vec3) {
+    let mut velocity = Vec3::new(MAX_VELOCITY * (velocity_rng.gen::<f32>() - 0.5), 0., 0.);
+
+    if let Some(waves) = waves {
+        // Step the movement and handle collisions as if the wave had been spawned at fixed time intervals
+        // and with dt-spaced frames of simulation
+        for _ in 0..(waves * (FIXED_TIMESTEP / dt).round() as usize) {
+            step_movement(&mut translation, &mut velocity, dt);
+            handle_collision(half_extents, &translation, &mut velocity);
+        }
+    }
+    (
+        Transform {
+            translation,
+            scale: Vec3::splat(BIRD_SCALE),
+            ..default()
+        },
```

Comment:
You can use this instead to make it a bit shorter

```suggestion
Transform::from_translation(translation).with_scale(Vec3::splat(BIRD_SCALE)),
```

Reply:
Done.